packages/dd-trace/src/ci-visibility/dynamic-instrumentation/worker/index.js

```diff
@@ -0,0 +1,88 @@
+const { parentPort } = require('worker_threads')
+// TODO: move debugger/devtools_client/session to common place
+const session = require('../../../debugger/devtools_client/session')
+// TODO: move debugger/devtools_client/snapshot to common place
+const { getLocalStateForCallFrame } = require('../../../debugger/devtools_client/snapshot')
+// TODO: move debugger/devtools_client/state to common place
+const {
+  findScriptFromPartialPath,
+  getStackFromCallFrames
+} = require('../../../debugger/devtools_client/state')
+const { BREAKPOINT_HIT, BREAKPOINT_SET } = require('../messages')
+const log = require('../../../log')
+
+let sessionStarted = false
+
+const breakpointIdToSnapshotId = new Map()
+const breakpointIdToProbe = new Map()
+
+session.on('Debugger.paused', async ({ params: { hitBreakpoints: [hitBreakpoint], callFrames } }) => {
+  const probe = breakpointIdToProbe.get(hitBreakpoint)
+  if (!probe) {
+    return session.post('Debugger.resume')
+  }
```

Comment:
Is this just an escape hatch in case the state is somehow messed up, is there a valid use-case for a breakpoint to exist, but no matching probe?

If this is just an escape hatch, I'd probably add a log line to log that you encountered this issue. But I'll leave that up you :)

Reply:
yeah it's a escape hatch. I encountered this situation while testing, but my code was all over the case, so I probably messed something up. Good idea to add a log line 👍

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/4826/commits/cd4b6fe40888043b45da080cb954447a25cd935a